nixos/modules/services/scheduling/scx.nix

```diff
@@ -0,0 +1,114 @@
+{
+  lib,
+  pkgs,
+  config,
+  ...
+}:
+let
+  cfg = config.services.scx;
+in
+{
+  options.services.scx = {
+    enable = lib.mkEnableOption null // {
+      description = ''
+        Whether to enable SCX service, a scheduler daemon with a wide variety of
+        scheduling algorithms to improve system performance.
+
+        ::: {.note}
+        This service requires a kernel with the Sched-ext feature. Generally, kernel version 6.12 and later are supported. See
+        [kernel documentation](https://www.kernel.org/doc/html/v6.12-rc1/scheduler/sched-ext.html#switching-to-and-from-sched-ext)
+        for details.
+        :::
+      '';
+    };
+
+    package = lib.mkOption {
```

Comment:
Use `mkPackageOption`.

Reply:
This was causing some issues. Can't override the description apparently.

And using it like:

```nix
lib.mkPackageOption pkgs [ "scx" "full" ] { }
```
the description to be: "which full package to use"

Reply:
OK.

Reply:
You can do `lib.mkPackageOption pkgs [ "scx" "full" ] // { … }`.

Reply:
I tried it before. NixOS manual fails to build.
```nix
lib.mkPackageOption pkgs [ "scx" "full" ] // {
      description = ''
        `scx` package to use. ....
      '';
    };
```

Reply:
Ah, sorry: the correct way is `lib.mkPackageOption pkgs "friendly name" { default = [ "scx" "full" ]; }`. Anyway, doesn’t matter.

Reply:
This just makes it more complex for no reason IMO.

Reply:
It would save several lines of code and docs here, but sure.